controllers/index.js

```diff
@@ -10,10 +10,47 @@ router.get('/', async (req, res, next) => {
     // Fetch templates
     const {templates} = await model.list(0, null);
     const filterOptions = {
-      sort: req.query.sort || 'views',
-      templateTypes: (req.query.templateTypes || 'all').split(','),
-      categories: (req.query.categories || 'all').split(',')
+      sort: (req.query.sort) ? req.query.sort.split(',') : ['views'],
+      templateTypes: (req.query.templateTypes) ? req.query.templateTypes.split(',') : ['all'],
+      categories: (req.query.categories) ? req.query.categories.split(',') : ['all']
     };
+    
+    // Sanitizing refining values
+    // Template Types
+    filterOptions.templateTypes.forEach(function(e,i){
+        if(Object.keys(enums.allowedFilterValues.templateTypes).indexOf(e)===-1) filterOptions.templateTypes.splice(i,1);
+    });
+    // User may remove the filter value manually and key will still exist, set default value
+    if(filterOptions.templateTypes.length===0) filterOptions.templateTypes = ['all'];
+
+    // Sort Types
+    filterOptions.sort.forEach(function(e,i){
+        if(Object.keys(enums.allowedFilterValues.sort).indexOf(e)===-1) filterOptions.sort.splice(i,1);
+    });
+    // User may remove the filter value manually and key will still exist, set default value
+    if(filterOptions.sort.length===0) filterOptions.sort = ['views'];
+
```

Comment:
Same comments about ES6 and the redundancy of the `filterOptions.sort.length` check.